cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -229,132 +247,182 @@ func randString8() string {
 	return string(b)
 }
 
-func (mgr *packetManagerRest) createNode(cloudinit, nodegroup string) {
+func (mgr *packetManagerRest) createNode(cloudinit, nodegroup string) error {
 	udvars := CloudInitTemplateData{
 		BootstrapTokenID:     os.Getenv("BOOTSTRAP_TOKEN_ID"),
 		BootstrapTokenSecret: os.Getenv("BOOTSTRAP_TOKEN_SECRET"),
 		APIServerEndpoint:    mgr.apiServerEndpoint,
 	}
-	ud := renderTemplate(cloudinit, udvars)
+
+	ud, err := renderTemplate(cloudinit, udvars)
+	if err != nil {
+		return fmt.Errorf("failed to create userdata from template: %w", err)
+	}
+
 	hnvars := HostnameTemplateData{
 		ClusterName: mgr.clusterName,
 		NodeGroup:   nodegroup,
 		RandString8: randString8(),
 	}
-	hn := renderTemplate(mgr.hostnamePattern, hnvars)
+	hn, err := renderTemplate(mgr.hostnamePattern, hnvars)
+	if err != nil {
+		return fmt.Errorf("failed to create hostname from template: %w", err)
+	}
+
+	if err := mgr.createDevice(hn, ud, nodegroup); err != nil {
+		return err
+	}
 
+	klog.Infof("Created new node on Packet.")
+
+	return nil
+}
+
+// createNodes provisions new nodes on packet and bootstraps them in the cluster.
+func (mgr *packetManagerRest) createNodes(nodegroup string, nodes int) error {
+	klog.Infof("Updating node count to %d for nodegroup %s", nodes, nodegroup)
+
+	cloudinit, err := base64.StdEncoding.DecodeString(mgr.cloudinit)
+	if err != nil {
+		log.Fatal(err)
+		return fmt.Errorf("Could not decode cloudinit script: %v", err)
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings
```suggestion
		return fmt.Errorf("could not decode cloudinit script: %v", err)
```